wagtail/blocks/base.py

```diff
@@ -269,6 +271,31 @@ def render(self, value, context=None):
 
         return mark_safe(render_to_string(template, new_context))
 
+    def get_preview_context(self, value, parent_context=None):
+        # We do not fall back to `get_context` here, because the preview context
+        # will be used for a complete view, not just the block. Instead, the
+        # default preview context uses `{% include_block %}`, which will use
+        # `get_context`.
+        return parent_context or {}
+
+    def get_preview_template(self, value=None, context=None):
+        # We do not fall back to `get_template` here, because the template will
+        # be used for a complete view, not just the block. In most cases, the
+        # block's template cannot stand alone for the preview, as it would be
+        # missing the necessary static assets.
+        #
+        # Instead, the default preview template uses `{% include_block %}`,
+        # which will use `get_template` if a template is defined.
+        return getattr(self.meta, "preview_template", None)
```

Comment:
I'm not entirely sure how we should handle this, i.e. whether the `get_preview_{template,context}` should be responsible for the entire view template and context or just the `{% block content %}` part. I'm leaning towards the former, as it allows you to customise exactly what the full HTML that will be loaded in the iframe is.

However, in most cases `get_template()` and `get_context()` are not useful for a complete view as it's likely missing the static assets, so it's not helpful to fallback to those methods.

In addition, the default preview template just does `{% include_block %}`, which goes through the `get_template()` and `get_context()` mechanism already. So, an easier way to configure the feature is just to override the preview template globally and add the static assets, leaving `get_preview_context` and `get_preview_template` as-is, and let the existing block rendering mechanism take place. Or, if devs are averse to overriding the template globally, have their own template that does the `{% include_block %}` (or not)  and override `get_preview_template` instead.